client/signup/steps/site-or-domain/index.jsx

```diff
@@ -36,15 +36,29 @@ class SiteOrDomain extends Component {
 		return isValidDomain && domain;
 	}
 
+	isLeanDomainSearch() {
+		const { signupDependencies } = this.props;
+		return (
+			!! signupDependencies &&
+			!! signupDependencies.refParameter &&
+			'leandomainsearch' === signupDependencies.refParameter
```

Comment:
Not a blocker: This could be simplified using optional chaining.
```suggestion
			'leandomainsearch' === signupDependencies?.refParameter
```